spot_driver/spot_driver/calibrated_reregistered_hand_camera_depth_publisher.py

```diff
@@ -0,0 +1,254 @@
+
+This script assumes you have a calibration generated with /spot_wrapper/spot_wrapper/calibration
+"""
+import argparse
+from typing import Optional
+
+import bdai_ros2_wrappers.process as ros_process
+import bdai_ros2_wrappers.scope as ros_scope
+import cv2
+import numpy as np
+import open3d as o3d
+import yaml
+from bdai_ros2_wrappers.context import wait_for_shutdown
+from cv_bridge import CvBridge
+from sensor_msgs.msg import Image
+
+
+class CalibratedReRegisteredHandCameraDepthPublisher:
+    def __init__(
+        self,
+        calibration_path: Optional[str] = None,
+        tag: str = "default",
+        robot_name: str = "spot",
+        topic_name: str = "depth_registered/hand_custom_cal/image",
+        undistort: bool = False,
+    ):
+        self.node = ros_scope.node()
+
+        if calibration_path is None:
+            self.node.get_logger().warning("No calibration path found, using default calibration")
+            self.calibration = {
+                "camera_matrix_depth": np.zeros(9).tolist(),
+                "dist_coeffs_depth": np.zeros(5).tolist(),
+                "camera_matrix_rgb": np.zeros(9).tolist(),
+                "dist_coeffs_rgb": np.zeros(5).tolist(),
+                "depth_t_rgb_R": np.zeros(9).tolist(),
+                "depth_t_rgb_T": np.zeros(3).tolist(),
+                "depth_image_dim": np.zeros(2).tolist(),
+                "rgb_image_dim": np.zeros(2).tolist(),
+            }
+        else:
+            self.calibration = extract_calibration_parameters(calibration_path=calibration_path, tag=tag)
+
+        self.calibration["robot_name"] = robot_name
+        self.calibration["topic_name"] = topic_name
+        self.calibration["undistort"] = undistort
+
+        for param_name, param_val in self.calibration.items():
+            self.node.get_logger().info(f"Setting the parameter {param_name} to a default value of {param_val}")
+            self.node.declare_parameter(param_name, param_val)
+            self.calibration[param_name] = self.node.get_parameter(param_name).value
+            if param_name in ["camera_matrix_depth", "camera_matrix_rgb", "depth_t_rgb_R"]:
+                self.calibration[param_name] = np.array(self.calibration[param_name]).reshape((3, 3))
+
+            elif "dim" not in param_name and "name" not in param_name and "undistort" not in param_name:
+                self.calibration[param_name] = np.array(self.calibration[param_name])
+
+        if self.calibration["undistort"]:
+            self.calculate_undistortion_parameters()
+
+        self.cv_bridge = CvBridge()
+
+        reregistered_depth_topic = f"/{self.calibration['robot_name']}/{topic_name}"
+        self.node.get_logger().info(f"Creating reregistered depth publisher to {reregistered_depth_topic}")
+        self.reregistered_depth_img_pub = self.node.create_publisher(Image, reregistered_depth_topic, 10)
+
+        raw_depth_topic = f"/{self.calibration['robot_name']}/depth/hand/image"
```

Comment:
The driver currently supports launching without a namespace -- if that's the case, the image topics look like `/camera/hand/image` instead of `/<RobotName>/camera/hand/image`. Can you account for that here? So if a robot name is not passed in, or is an empty string, it follows this same convention?